01.Importing all the required libraires at once.

In [1]:
# These libraries are sufficient for our Data analysis section.
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

#Importing the dataset.

In [2]:
df = pd.read_csv(r'C:\Users\Nikhil Pandey\OneDrive\Desktop\PROGRAM\Projects\Heart_desease\data\raw\diabetes_raw.csv')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101766 entries, 0 to 101765
Data columns (total 48 columns):
 #   Column                    Non-Null Count   Dtype 
---  ------                    --------------   ----- 
 0   race                      99493 non-null   object
 1   gender                    101766 non-null  object
 2   age                       101766 non-null  object
 3   weight                    3197 non-null    object
 4   admission_type_id         101766 non-null  int64 
 5   discharge_disposition_id  101766 non-null  int64 
 6   admission_source_id       101766 non-null  int64 
 7   time_in_hospital          101766 non-null  int64 
 8   payer_code                61510 non-null   object
 9   medical_specialty         51817 non-null   object
 10  num_lab_procedures        101766 non-null  int64 
 11  num_procedures            101766 non-null  int64 
 12  num_medications           101766 non-null  int64 
 13  number_outpatient         101766 non-null  int64 
 14  numb

In [4]:
print(df.columns.tolist()) # Extracting titlecims

['race', 'gender', 'age', 'weight', 'admission_type_id', 'discharge_disposition_id', 'admission_source_id', 'time_in_hospital', 'payer_code', 'medical_specialty', 'num_lab_procedures', 'num_procedures', 'num_medications', 'number_outpatient', 'number_emergency', 'number_inpatient', 'diag_1', 'diag_2', 'diag_3', 'number_diagnoses', 'max_glu_serum', 'A1Cresult', 'metformin', 'repaglinide', 'nateglinide', 'chlorpropamide', 'glimepiride', 'acetohexamide', 'glipizide', 'glyburide', 'tolbutamide', 'pioglitazone', 'rosiglitazone', 'acarbose', 'miglitol', 'troglitazone', 'tolazamide', 'examide', 'citoglipton', 'insulin', 'glyburide-metformin', 'glipizide-metformin', 'glimepiride-pioglitazone', 'metformin-rosiglitazone', 'metformin-pioglitazone', 'change', 'diabetesMed', 'readmitted']


In [ ]:
for col in df.columns:
    print(f"\n{'='*55}")
    print(f"COLUMN : {col}")
    print(f"dtype  : {df[col].dtype}")
    print(f"missing: {df[col].isnull().sum()} | '?' values: {(df[col] == '?').sum()}")
    print(f"unique : {df[col].nunique()}")
    print(f"values : {df[col].value_counts().head(5).to_dict()}")
    print(f"{'='*55}")
    print(f"")


COLUMN : race
dtype  : object
missing: 2273 | '?' values: 0
unique : 5
values : {'Caucasian': 76099, 'AfricanAmerican': 19210, 'Hispanic': 2037, 'Other': 1506, 'Asian': 641}


COLUMN : gender
dtype  : object
missing: 0 | '?' values: 0
unique : 3
values : {'Female': 54708, 'Male': 47055, 'Unknown/Invalid': 3}


COLUMN : age
dtype  : object
missing: 0 | '?' values: 0
unique : 10
values : {'[70-80)': 26068, '[60-70)': 22483, '[50-60)': 17256, '[80-90)': 17197, '[40-50)': 9685}


COLUMN : weight
dtype  : object
missing: 98569 | '?' values: 0
unique : 9
values : {'[75-100)': 1336, '[50-75)': 897, '[100-125)': 625, '[125-150)': 145, '[25-50)': 97}


COLUMN : admission_type_id
dtype  : int64
missing: 0 | '?' values: 0
unique : 8
values : {1: 53990, 3: 18869, 2: 18480, 6: 5291, 5: 4785}


COLUMN : discharge_disposition_id
dtype  : int64
missing: 0 | '?' values: 0
unique : 26
values : {1: 60234, 3: 13954, 6: 12902, 18: 3691, 2: 2128}


COLUMN : admission_source_id
dtype  : int64
missing: 0 | '

# My personal depiction of the dataset:

The insights which I can exclaim using the dataframe after spending sometime on EDA:
* **Race**: Details are given on the basis of race of the people which also consists some missing values in the column. It has  2273 missing values. Categorized as :'Caucasian', 'AfricanAmerican', nan, 'Other', 'Asian', 'Hispanic


* **gender**: Has normal male and female categories will be better for feature enginering.  Female             54708
Male               47055
Unknown/Invalid        3


* **age**: Lies normal in range.


* **weight**: column exists but have minimal value in it with the information for this whole dataset.

* **Admission type id**: It seems like have classification from 1 to 8.

* **discharge_disposition_id** :

* **admission_source_id** :

* **time_in_hospital** :

* **payer_code** : It has some mising values :

[nan, 'MC', 'MD', 'HM', 'UN', 'BC', 'SP', 'CP', 'SI', 'DM', 'CM',
       'CH', 'PO', 'WC', 'OT', 'OG', 'MP', 'FR'

payer_code
False    61510
True     40256

* **medical_specialty** : It has 72 different categorical data and each is a special medical condition.
49949 missing values


* **num_lab_procedures** :


* **num_procedures** :
* **num_medications** :
* **number_outpatient** :
* **number_emergency** :
* **number_inpatient** :
* **diag_1** :
* **diag_2** :
* **diag_3** :
* **number_diagnoses** :
* **max_glu_serum** :
* **A1Cresult** :
* **metformin** :
* **repaglinide** :
* **nateglinide** :
* **chlorpropamide** :
* **glimepiride** :
* **acetohexamide** :
* **glipizide** :
* **glyburide** :
* **tolbutamide** :
* **pioglitazone** :
* **rosiglitazone** :
* **acarbose** :
* **miglitol** :
* **troglitazone** :
* **tolazamide** :
* **examide** :
* **citoglipton** :
* **insulin** :
* **glyburide-metformin** :
* **glipizide-metformin** :
* **glimepiride-pioglitazone** :
* **metformin-rosiglitazone** :
* **metformin-pioglitazone** :
* **change** :
* **diabetesMed** :
* **readmitted** :



In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101766 entries, 0 to 101765
Data columns (total 48 columns):
 #   Column                    Non-Null Count   Dtype 
---  ------                    --------------   ----- 
 0   race                      99493 non-null   object
 1   gender                    101766 non-null  object
 2   age                       101766 non-null  object
 3   weight                    3197 non-null    object
 4   admission_type_id         101766 non-null  int64 
 5   discharge_disposition_id  101766 non-null  int64 
 6   admission_source_id       101766 non-null  int64 
 7   time_in_hospital          101766 non-null  int64 
 8   payer_code                61510 non-null   object
 9   medical_specialty         51817 non-null   object
 10  num_lab_procedures        101766 non-null  int64 
 11  num_procedures            101766 non-null  int64 
 12  num_medications           101766 non-null  int64 
 13  number_outpatient         101766 non-null  int64 
 14  numb